op-bindings/etherscan-client/etherscan_client.go

```diff
@@ -0,0 +1,247 @@
+package etherscanclient
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type EtherscanClient struct {
+	apiMaxRetries int
+	apiRetryDelay time.Duration
+	baseUrls      map[int]string
+	apiKeys       map[int]string
+}
+
+type etherscanApiResponse struct {
+	Status  string      `json:"status"`
+	Message string      `json:"message"`
+	Result  interface{} `json:"result"`
+}
+
+type etherscanRpcResponse struct {
+	JsonRpc string      `json:"jsonrpc"`
+	Id      int         `json:"id"`
+	Result  interface{} `json:"result"`
+}
+
+type txInfo struct {
+	Input string `json:"input"`
+}
+
+func validateChainId(chainId int) error {
+	switch chainId {
+	case 1, 10:
+	default:
+		return fmt.Errorf("unsupported source chain ID: %d", chainId)
+	}
+
+	return nil
+}
+
+func NewEtherscanClient(
```

Comment:
This is verbose - the package name already has `etherscan` in it, you can just do `etherscan.Client`